paper.tex

```diff
@@ -175,44 +169,44 @@ \subsubsection{Apache Spark}
 child, RDD through a narrow or wide dependency. A narrow dependency signifies 
 that the child is only dependent on a single parent partition, whereas a child 
 RDD is dependent on all parent partitions in a wide dependency. Examples of 
-spark transformations include map, filter and join. To materialize an RDD, an
+Spark transformations include map, filter and join. To materialize an RDD, an
 action must be performed, such as a reduce or a collect.
 
 All actions and wide dependencies require a shuffle -- Spark's most costly
 operation. Every shuffle begins with each map task saving its data to local
 files to be fetched by the shuffle operation. The shuffle operation then 
 redistributes the data across the partitions as requested. A shuffle marks a 
 stage boundary in Spark. In other words, a shuffle will not begin until all 
-dependent map tasks have completed.
+dependent map tasks \todo{shouldn't it be 'actions'?} have completed.
```

Comment:
Yes, but no. Actions materialized the RDDs, but shuffles still wait for its dependent transformations to finalize before the _reduce_ commences (i wrote 'shuffle' accidentally)